youtube_dl/extractor/instagram.py

```diff
@@ -245,117 +351,65 @@ class InstagramUserIE(InfoExtractor):
-
-                comment_count = get_count('to_comment')
-                like_count = get_count('preview_like')
-                view_count = int_or_none(node.get('video_view_count'))
-
-                info.update({
-                    'description': description,
-                    'thumbnail': thumbnail,
-                    'timestamp': timestamp,
-                    'comment_count': comment_count,
-                    'like_count': like_count,
-                    'view_count': view_count,
-                })
-
-                yield info
-
-            page_info = media.get('page_info')
-            if not page_info or not isinstance(page_info, dict):
-                break
-
-            has_next_page = page_info.get('has_next_page')
-            if not has_next_page:
-                break
-
-            cursor = page_info.get('end_cursor')
-            if not cursor or not isinstance(cursor, compat_str):
-                break
+        return self.playlist_result(
+            extract_graphql(self, data,
+                            _id=username,
+                            query_hash='42323d64886122307be10013ad2dcc44',
+                            query_vars={'id': data['entry_data']['ProfilePage'][0]['graphql']['user']['id']},
+                            timeline_fn=lambda j:
+                                j['data']['user']['edge_owner_to_timeline_media']
+                            ),
+            username, username)
+
+
+class InstagramTagIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?instagram\.com/explore/tags/(?P<tag>[^/]+)'
+    IE_DESC = 'Instagram search query'
+    IE_NAME = 'instagram:tag'
+    _TEST = {
+        'url': 'https://instagram.com/explore/tags/lolcats',
+        'info_dict': {
+            'id': 'lolcats',
+            'title': 'lolcats',
+        },
+        'playlist_count': 50,
+        'params': {
+            'extract_flat': True,
+            'skip_download': True,
+            'playlistend': 50,
+        }
+    }
 
     def _real_extract(self, url):
-        username = self._match_id(url)
+        m = self._VALID_URL_RE.search(url)
+        tag = m.group('tag')
+        assert tag
 
-        webpage = self._download_webpage(url, username)
+        webpage = self._download_webpage(url, tag)
 
         data = self._parse_json(
             self._search_regex(
                 r'sharedData\s*=\s*({.+?})\s*;\s*[<\n]', webpage, 'data'),
-            username)
+            tag)
 
         return self.playlist_result(
-            self._entries(data), username, username)
+            extract_graphql(self, data,
+                            _id=tag,
+                            query_hash='f92f56d47dc7a55b606908374b43a314',
+                            query_vars={'tag_name': tag},
+                            timeline_fn=lambda j:
+                                j['data']['hashtag']['edge_hashtag_to_media']
+                            ),
+            tag, tag)
```

Comment:
DRY.

Reply:
Where is the repetition? All of the call arguments are different between the two classes.